src/go/rpk/pkg/cli/cmd/wasm/common.go

```diff
@@ -54,3 +56,19 @@ func ExistingTopic(admin sarama.ClusterAdmin, topic string) (bool, error) {
 		return false, err
 	}
 }
+
+/**
+Create coprocessor message
+*/
+func CreateCoprocessorMessage(
```

Comment:
This is kind of general, all it does it wrap up the logic of creating a kafka record. How about something more high level:
```
type WasmRemoveEvent struct {
  uint64 id;
  string desc
};
type WasmDeployEvent struct {
  uint64 id
  string desc
  string filename
  iobuf src_code
};

func CreateCoprocessorMessage(WasmRemoveEvent record) .....
func CreateCoprocessorMessage(WasmDeployEvent record) .....
```

Reply:
Umm but it is the same, no? I need to implement 2 functions more, and right now I have those process on their functions (deploy, remove)

Reply:
Its not the same because the methods accept a type. Making things explicitly typed ensures correctness. For example with the old API someone could create a 'remove' event but pass headers that appear as a deploy event. Or the other way around. Someone could be making a remove event, but pass a value for 'message' and that would fail validation on the redpanda side. You should be building the headers that correspond to the type of message your creating. Not accepting that they are correct from the call-ee

Reply:
Got it, thanks for explaining